src/yunohost/data_migrations/0009_setup_group_permission.py

```diff
@@ -0,0 +1,142 @@
+import yaml
+import time
+import os
+import shutil
+
+from moulinette import m18n
+from moulinette.core import init_authenticator
+from yunohost.utils.error import YunohostError
+from moulinette.utils.log import getActionLogger
+
+from yunohost.tools import Migration
+from yunohost.utils.filesystem import free_space_in_directory, space_used_by_directory
+from yunohost.user import user_list, user_group_add, user_group_update
+from yunohost.app import app_setting, app_list
+from yunohost.service import service_regen_conf
+from yunohost.permission import permission_add, permission_sync_to_user
+from yunohost.user import user_permission_add
+
+logger = getActionLogger('yunohost.migration')
+
+###################################################
+# Tools used also for restoration
+###################################################
+
+def migrate_LDAP_db(auth):
```

Comment:
You might want to move this inside the migration ? I guess you put it outside because you wanted to access it by just loading the module but maybe with `_get_migration_by_name` that should be okay :stuck_out_tongue_winking_eye: